homeassistant/components/template/entity.py

```diff
@@ -0,0 +1,66 @@
+"""Template entity base class."""
+
+from collections.abc import Sequence
+from typing import Any
+
+from homeassistant.core import Context, HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.script import Script, _VarsType
+from homeassistant.helpers.template import TemplateStateFromEntityId
+
+
+class AbstractTemplateEntity(Entity):
+    """Actions linked to a template entity."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the entity."""
+
+        self.hass = hass
+        self._action_scripts: dict[str, Script] = {}
+
+    @property
+    def referenced_blueprint(self) -> str | None:
+        """Return referenced blueprint or None."""
+        raise NotImplementedError
+
+    @callback
+    def _render_script_variables(self) -> dict:
+        """Render configured variables."""
+        raise NotImplementedError
+
+    def add_script(
+        self,
+        script_id: str,
+        config: Sequence[dict[str, Any]],
+        name: str,
+        domain: str,
+    ):
+        """Add an action script."""
+
+        # Cannot use self.hass because it may be None in child class
```

Comment:
This comment is confusing since we now access `self.hass`. I'd move the comment to the init method and explain why we need to explicitly pass in `hass`.